atomic_reactor/plugins/pre_pull_base_image.py

```diff
@@ -97,122 +102,65 @@ def run(self):
                                  current_platform, image)
             else:
                 self.log.info("Replacing image '%s' with '%s'", image, image_with_digest)
                 image = image_with_digest
 
-            if self.check_platforms:
-                new_arch_image = self._get_image_for_different_arch(image, current_platform)
-                if new_arch_image:
-                    image = new_arch_image
-
-            if self.inspect_only:
-                new_image = image
-            else:
-                new_image = self._pull_and_tag_image(image, build_json, str(nonce))
+            if not self.inspect_only:
+                image = self._pull_and_tag_image(image, build_json, str(nonce))
             self.workflow.builder.recreate_parent_images()
-            self.workflow.builder.parent_images[parent] = new_image
+            self.workflow.builder.parent_images[parent] = image
 
             if is_base_image:
                 if organization:
                     # we want to be sure we have original_base_image enclosed as well
                     self.workflow.builder.original_base_image.enclose(organization)
                 self.workflow.builder.set_base_image(
-                    str(new_image), insecure=self.parent_registry_insecure,
+                    str(image), insecure=self.parent_registry_insecure,
                     dockercfg_path=self.parent_registry_dockercfg_path
                 )
         self.workflow.builder.parents_pulled = not self.inspect_only
         self.workflow.builder.base_image_insecure = self.parent_registry_insecure
 
-    def _get_image_with_digest(self, image, platform):
-        parents_digests = self.workflow.builder.parent_images_digests
-
+    def _get_image_with_digest(self, image):
+        image_str = image.to_str()
         try:
-            new_image = ImageName.parse(parents_digests.get_image_platform_digest(image, platform))
+            digest = self.workflow.builder.parent_images_metadata[image_str]['digests']['sha256sum']
         except KeyError:
             return None
 
-        return new_image
-
-    def _load_parent_images_digests(self, images_digests):
-        assert isinstance(images_digests, dict)
-        parents_digests = self.workflow.builder.parent_images_digests
-        parents_digests.update_from_dict(images_digests)
-
-    def _collect_image_digests(self, image):
-        parents_digests = self.workflow.builder.parent_images_digests
-
-        if image in parents_digests:
-            # we already have recorded digests for the image, keep it the same
-            # to prevent race conditions at workers
-            return
+        image_name = image.to_str(tag=False)
+        new_image = '{}@sha256:{}'.format(image_name, digest)
+        return ImageName.parse(new_image)
 
+    def _store_manifest_digest(self, image):
+        """Store manifest list or v2 schema 2 manifest digest """
```

Comment:
I'm not very comfortable with mixing 2 kinds of information into the same data structure. Could we separate v2 digest and list digest at some level?

Reply:
we aren't exactly missing different kind of information as we have dict :
{'digests': manifest_digest, 'media_type': media_type}

but we should probably reflect that in that comment

Reply:
we are mixing, once it stores manifest list, then it may store v2 manifest digest. If we would need to store both, this method must be rewritten completely. But can stay as is it has at least identification what kind is stored there

Reply:
mmmm, when we will use images build with group manifests enabled,
we will always find v2_list for all parents (which were build with group manifests enabled)

so unless somebody is mixing some very old build, which had only v2s2, with builds where group manifests were enabled, we will always have there either only all manifest_lists or only v2s2 digests

Reply:
but that comment could be something like "store digest and media types for v2s2 or v2_list"

Reply:
It's not really mixed. They're all manifest digests. Manifests come in different flavours. It's the same sort of idea as the platform specifier: a digest and a media type.

Reply:
I did change the docstring here. The method is slightly different from when the comment was added